### PR TITLE
Pure R iRODS demonstration

### PR DESCRIPTION
**Pull request by MartinSchobben**
The functions `use_irods_demo()` and `stop_irods_demo()` are meant for convenience during demonstration and exploration of the package's functionality. Now everything can be contained in an R script. This can be convenient when for example doing a demo (e.g. at the iRODS UGM). Now the repo `irods_demo` is also contained as an git submodule within the `rirods` repo. The problem might be that this causes problems during CRAN submission as this cannot be tested on e.g. Windows and that it tampers with the users system (i.e. running docker containers). But there is precedence for that as  the CRAN `rhub` package does something similar: https://r-hub.github.io/rhub/articles/local-debugging.html.

### CODE REVIEW

**Comment by trel** [comment]
happy to merge if this is behaving as desired.    i don't know about the CRAN process - so have no additional insight.

File: R/test-helpers.R
Comment (montesmariana) [inline R/test-helpers.R:1]: I'm a bit worried that it might be confusing to have an R file (not test file) that starts with 'test-'

**Review by montesmariana — APPROVED** [review]
As far as I understand it, I think it looks good and I can't wait to use it :) I think it's a great addition to the package

File: R/test-helpers.R
Comment (MartinSchobben) [reply R/test-helpers.R:1]: I'll rename to dev-helpers

Comment (MartinSchobben) [reply R/test-helpers.R:1]: But I will do the rename then in the next PR, which follows right after this one.

**Comment by MartinSchobben** [comment]
@trel can you merge it? Thanks!

**Comment by trel** [comment]
merging!

**Pull request merged**